diff/20170703/20170703.3.sql

```diff
@@ -0,0 +1,45 @@
+-- Moving from 2d to 3d geometry; LV95=2056 (av03=21781)
+-- only tables with real 3d geometry
+------------------------------------
+
+-- FEHLER:  kann Tabelle qgep.od_wastewater_structure Spalte detail_geometry3d_geometry nicht löschen, weil andere Objekte davon abhängen
+-- DETAIL:  Sicht qgep.vw_channel hängt von Tabelle qgep.od_wastewater_structure Spalte detail_geometry3d_geometry ab
+-- Sicht qgep.vw_discharge_point hängt von Tabelle qgep.od_wastewater_structure Spalte detail_geometry3d_geometry ab
+-- Sicht qgep.vw_manhole hängt von Tabelle qgep.od_wastewater_structure Spalte detail_geometry3d_geometry ab
+-- Sicht qgep.vw_special_structure hängt von Tabelle qgep.od_wastewater_structure Spalte detail_geometry3d_geometry ab
+-- HINT:  Verwenden Sie DROP ... CASCADE, um die abhängigen Objekte ebenfalls zu löschen.
+
+
+--od_wastewater_structure
+ALTER TABLE qgep.od_wastewater_structure DROP COLUMN detail_geometry3d_geometry CASCADE;
+ALTER TABLE qgep.od_wastewater_structure ADD COLUMN detail_geometry3d_geometry geometry(CurvePolygonZ, :SRID);
+UPDATE qgep.od_wastewater_structure SET detail_geometry3d_geometry = st_force3d(detail_geometry_geometry);
+ALTER TABLE qgep.od_wastewater_structure DROP COLUMN detail_geometry_geometry CASCADE;
```

Comment:
I would really avoid deltete cascade in update script. Better write manually all the elements, that would avoid removing user extra in the db without him noticing.